File: rust/src/models/introspection_request.rs
// this file is @generated
use serde::{Deserialize, Serialize};

/// Token introspection request
#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct IntrospectionRequest {
    /// The token to introspect
    pub token: String,
}

impl IntrospectionRequest {
    pub fn new(token: String) -> Self {
        Self { token }
    }
}

File: typescript/src/models/introspectionRequest.ts
// this file is @generated
/** Token introspection request */
export interface IntrospectionRequest {
  /** The token to introspect */
  token: string;
}

export const IntrospectionRequestSerializer = {
  _fromJsonObject(object: any): IntrospectionRequest {
    return {
      token: object["token"],
    };
  },

  _toJsonObject(self: IntrospectionRequest): any {
    return {
      token: self.token,
    };
  },
};
